common/djangoapps/student/models.py

```diff
@@ -1442,6 +1442,28 @@ def enrollments_for_user(cls, user):
         return cls.objects.filter(user=user, is_active=1)
 
     @classmethod
+    def preloaded_enrollments_for_user(cls, user):
+        """
```

Comment:
The pre-loading is relatively expensive (often more so than the initial load of enrollments), because you basically have to do a fat select-in query. That being said, my first stab at this did use an optional param as @pwnage101 suggested. The reason I switched to this approach is because we need to pull the `CourseEnrollment` model instances out of the `QuerySet` iterator and into a list of some sort in order to collect all the IDs and attach `CourseOverview` objects with one batch query. That would mean changing the return type, and other callers of this method expect a QuerySet (because they chain another option or two at the end).

I didn't want to have the same method return QuerySet or a list depending on the param, and I wanted to keep the scope of these changes as small as possible. So I thought the least risky thing to do was to just make a new method altogether, and have the dashboard view use it.

Definitely open to any naming suggestions. I'm not crazy about the naming either. I originally meant it to also do the select_related on the user in this method, but I decided to add that as just a baseline thing (along the line of @robrap's thinking) in #14545.